### PR TITLE
Compact cluster support

### PR DESCRIPTION
**Pull request by christianh814**
Making edits for compact cluster support. This endeavours to keep the playbook "backwards compatible". The only requirement to create a compact cluster is to _*NOT*_ define `workers`

Example for DHCP:

```

---
disk: vda
helper:
  name: "helper"
  ipaddr: "192.168.7.77"
dns:
  domain: "example.com"
  clusterid: "ocp4"
  forwarder1: "8.8.8.8"
  forwarder2: "8.8.4.4"
dhcp:
  router: "192.168.7.1"
  bcast: "192.168.7.255"
  netmask: "255.255.255.0"
  poolstart: "192.168.7.10"
  poolend: "192.168.7.30"
  ipid: "192.168.7.0"
  netmaskid: "255.255.255.0"
bootstrap:
  name: "bootstrap"
  ipaddr: "192.168.7.20"
  macaddr: "52:54:00:60:72:67"
masters:
  - name: "master0"
    ipaddr: "192.168.7.21"
    macaddr: "52:54:00:e7:9d:67"
  - name: "master1"
    ipaddr: "192.168.7.22"
    macaddr: "52:54:00:80:16:23"
  - name: "master2"
    ipaddr: "192.168.7.23"
    macaddr: "52:54:00:d5:1c:39"
```

This PR also addresses issues #125 #125 #106 #92

### CODE REVIEW

**Comment by christianh814** [comment]
@Prajyot-Parab @therevoman @bpradipt 

This is my first attempt to support compact clusters. Can you guys please check...

1. If the playbook works "as normal" with 3 masters and 2 workers
2. Check to see if "compact clusters" works (3 nodes acting as master/worker)

Tested it myself, but would like verification.

File: tasks/validate_host_names.yaml [hunk 1]
```diff
@@ -1,24 +1,10 @@
...
-  register: val1
-  ignore_errors: True
-
-- fail:
-    msg: "Please revise your vars.yaml file invalid characters in hostnames"
-  when:
-    (val0.failed is defined) or
-    (val1.failed is defined)
+  - "{{ workers | default('') }}"
+  register: val0
```
Comment (dlbewley) [inline tasks/validate_host_names.yaml:10]: i think you can drop the result of this task. i.e. no `register`

File: tasks/validate_host_names.yaml [hunk 2]
```diff
@@ -1,24 +1,10 @@
 - fail:
```
Comment (dlbewley) [inline tasks/validate_host_names.yaml:1]: i like to name my tasks so there is some context in the output eg. `name: Validate values for DNS compatibility`

File: tasks/validate_host_names.yaml (hunk 1 above)
Comment (christianh814) [reply tasks/validate_host_names.yaml]: Yup, you're right not needed. Pushed update.

File: tasks/validate_host_names.yaml (hunk 2 above)
Comment (christianh814) [reply tasks/validate_host_names.yaml]: ACK, good catch

**Pull request merged**